lib/modules/dosomething/dosomething_northstar/dosomething_northstar.auth.inc

```diff
@@ -160,3 +166,33 @@ function dosomething_northstar_user_logout($account) {
 
   drupal_goto($is_openid_login ? $northstar_logout_url : '/');
 }
+
+/**
+ * Submit handler for handling "classic" logins of accounts that do not have
+ * local passwords, using Northstar's `v1/auth/verify` endpoint.
+ *
+ * @param $form
+ * @param $form_state
+ */
+function dosomething_northstar_fallback_classic_login(&$form, &$form_state) {
+  // If the account does not have a local password, verify using Northstar
+  $northstar_user = dosomething_northstar_verify_user([
+    'username' => $form_state['input']['name'],
+    'password' => $form_state['input']['pass'],
+  ]);
+
+  // If successful response & ID field is given, the credentials worked.
+  if (! empty($northstar_user['data']['id'])) {
+    $sub = $northstar_user['data']['id'];
```

Comment:
Named for the [`sub` JWT claim](https://github.com/DoSomething/northstar/blob/dev/documentation/authentication.md#access-tokens), since the OpenID Connect method below refers to it that way.